### ISSUE TITLE
Performance bottlenecks

### ISSUE THREAD

**Issue by carlosfranzreb** [carlosfranzreb/train_vocoder#1]
These are the bottlenecks in the training script, according to a run with the [line profiler](https://github.com/pyutils/line_profiler):

1. Forward pass of validation batches (30%): mainly because the batch size is 1.
2. Backward pass of training batches (28%): could be improved with mixed precision and gradient accumulation.
